swarm/network/stream/peer.go

```diff
@@ -182,15 +183,49 @@ func (p *Peer) SendOfferedHashes(s *server, f, t uint64) error {
 	defer sp.Finish()
 
 	defer metrics.GetOrRegisterResettingTimer("send.offered.hashes", nil).UpdateSince(time.Now())
+	var err error
+	if s.stream.Live {
```

Comment:
@zelig @justelad I realize that we have bugs in `setNextBatch` and that's why you modified it here, but this PR has no description what-so-ever, so it is hard to follow what you are doing or why you are doing it.

Please submit smaller PRs with meaningful description when making changes.

@zelig I am writing this as a response to our conversation earlier today, when you suggested we should continue from your recent changes, and I assume you meant this PR.